hal/network/lwip/wiznet/wiznetif_config.cpp

```diff
@@ -0,0 +1,206 @@
+}
+
+void WizNetifConfig::logWizNetifConfigData(WizNetifConfigData& configData) {
+    WZCFG_LOG_DEBUG(INFO, "wizNetifConfigData size:%u ver:%u cs_pin:%u reset_pin:%u int_pin:%u",
+            configData.size,
+            configData.version,
+            configData.cs_pin,
+            configData.reset_pin,
+            configData.int_pin);
+}
+
+int WizNetifConfig::saveWizNetifConfigData() {
+    WizNetifConfigData tempData = {};
+    tempData.size = sizeof(WizNetifConfigData);
+    tempData.version = WIZNETIF_CONFIG_DATA_VERSION;
+    int result = SystemCache::instance().get(SystemCacheKey::WIZNET_CONFIG_DATA, (uint8_t*)&tempData, sizeof(tempData));
+    if (result != std::min(tempData.size, wizNetifConfigData_.size) ||
+            memcmp(&tempData, &wizNetifConfigData_, std::min(tempData.size, wizNetifConfigData_.size)) != 0) {
+        WZCFG_LOG_DEBUG(INFO, "Saving cached wizNetifConfigData, size: %d", wizNetifConfigData_.size);
+        logWizNetifConfigData(wizNetifConfigData_);
+        result = SystemCache::instance().set(SystemCacheKey::WIZNET_CONFIG_DATA, (uint8_t*)&wizNetifConfigData_, sizeof(wizNetifConfigData_));
+    }
+    return (result < 0) ? result : SYSTEM_ERROR_NONE;
+}
+
+int WizNetifConfig::recallWizNetifConfigData() {
+    WizNetifConfigData tempData = {};
+    tempData.size = sizeof(WizNetifConfigData);
+    tempData.version = WIZNETIF_CONFIG_DATA_VERSION;
+    int result = SystemCache::instance().get(SystemCacheKey::WIZNET_CONFIG_DATA, (uint8_t*)&tempData, sizeof(tempData));
+    if (result != std::min(tempData.size, wizNetifConfigData_.size)) {
+        return SYSTEM_ERROR_NOT_FOUND;
+    }
+    WZCFG_LOG_DEBUG(INFO, "Recalling cached wizNetifConfigData, size: %d", result);
+    logWizNetifConfigData(tempData);
+    memcpy(&wizNetifConfigData_, &tempData, sizeof(tempData));
+    return SYSTEM_ERROR_NONE;
+}
+
+// int WizNetifConfig::deleteWizNetifConfigData() {
+//     WZCFG_LOG_DEBUG(INFO, "Deleting cached wizNetifConfigData");
+//     int result = SystemCache::instance().del(SystemCacheKey::WIZNET_CONFIG_DATA);
+//     return (result < 0) ? result : SYSTEM_ERROR_NONE;
+// }
+
+int WizNetifConfig::setConfigData(const WizNetifConfigData* configData) {
+    WizNetifConfigLock lk;
+
+    if (!initialized_) {
+        init();
+    }
+
+    CHECK_TRUE(configData, SYSTEM_ERROR_INVALID_ARGUMENT);
+    CHECK_TRUE(validateWizNetifConfigData(configData) == SYSTEM_ERROR_NONE, SYSTEM_ERROR_INVALID_ARGUMENT);
+
+    memcpy(&wizNetifConfigData_, configData, std::min(configData->size, wizNetifConfigData_.size));
+    logWizNetifConfigData(wizNetifConfigData_);
+    saveWizNetifConfigData();
+
+    return SYSTEM_ERROR_NONE;
+}
+
+int WizNetifConfig::getConfigData(WizNetifConfigData* configData) {
+    WizNetifConfigLock lk;
+
+    if (!initialized_) {
+        init();
+    }
+
+    CHECK_TRUE(configData, SYSTEM_ERROR_INVALID_ARGUMENT);
+
+    recallWizNetifConfigData();
+
+    if (configData->size == 0) {
```

Comment:
It's the caller responsibility to set `size` correctly, so ideally this should be an error condition.